pkg/agents/randomAgent/agent.go

```diff
@@ -4,16 +4,20 @@ import (
 	"math/rand"
 
 	"github.com/SOMAS2021/SOMAS2021/pkg/infra"
-	"github.com/SOMAS2021/SOMAS2021/pkg/utils/globalTypes/agent"
+	"github.com/SOMAS2021/SOMAS2021/pkg/utils/globalTypes/world"
+	log "github.com/sirupsen/logrus"
 )
 
 type CustomAgentRandom struct {
 	*infra.Base
 	// new params
 }
 
-func New(baseAgent *infra.Base) (agent.Agent, error) {
-	//create other parameters
+func New(world world.World, agentType int, agentHP int, agentFloor int, id string) (infra.Agent, error) {
```

Comment:
I don't think we should be passing the HP, Floor, and world to custom agents when they initialise a base agent for inheritance. It's counter-intuitive in my opinion because it means someone can just override their agent's starting HP or floor or even their base agent parameters. It also gives custom agents direct access to the world, which exposes the state of the tower, and deprecates the use of a base agent for state abstraction